src/Files.App/Views/Shells/BaseShellPage.cs

```diff
@@ -493,10 +493,12 @@ public void SubmitSearch(string query)
 				SearchQuery = query,
 			};
 
-			if (this is ColumnShellPage)
+			var layout = InstanceViewModel.FolderSettings.GetLayoutType(FilesystemViewModel.WorkingDirectory);
+
+			if (layout == typeof(ColumnsLayoutPage))
 				NavigateToPath(FilesystemViewModel.WorkingDirectory, typeof(DetailsLayoutPage), args);
 			else
-				ItemDisplay.Navigate(InstanceViewModel.FolderSettings.GetLayoutType(FilesystemViewModel.WorkingDirectory), args);
+				ItemDisplay.Navigate(layout, args);
```

Comment:
```suggestion
				NavigateToPath(FilesystemViewModel.WorkingDirectory, layout, args);
```
We should always call `NavigateToPath` at page transitions.

Reply:
What's the difference?

Reply:
The processing is specific to the implementation class of IShellPage.
Especially in CloumnShellPage, calling Navigate directly may cause unexpected behavior.